language-server-commons/src/main/java/com/palantir/ls/util/FileBackedContentsManager.java

```diff
@@ -74,7 +76,8 @@ public Path getDestination() {
     /**
      * Applies the given changes to the destination file. Does not handle intersecting ranges in the changes.
      */
-    public synchronized void applyChanges(List<TextDocumentContentChangeEvent> contentChanges) throws IOException {
+    @Override
+    public synchronized void applyChanges(List<TextDocumentContentChangeEvent> contentChanges) {
```

Comment:
shall we have an abstract class here since there is a fair amount of duplicate code?